### PR TITLE
chore: Remove axios dependency

### PR DESCRIPTION
**Pull request by atdrago**
This PR removes the axios (and moxios) dependencies. I also added `typescript` as a dev dependency because it wasn't clear which version this project was using.

This PR is related to https://github.com/Logflare/pino-logflare/pull/45

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -10,7 +10,6 @@
   "license": "MIT",
   "dependencies": {
     "@types/lodash": "^4.14.153",
```
Comment (atdrago) [inline package.json:12]: `lodash` is used in `src/typecasting.ts`. Nothing from that file is imported into `src/main.ts`, so `lodash` wouldn't get added to bundles unless they import from `src/typecasting.ts` directly.

It seems like the typecasting stuff is either a planned feature, or one that was removed, but I can't figure it out from the commit history. @chasers Do you know?

File: package.json [hunk 2]
```diff
@@ -24,8 +23,8 @@
     "@types/node": "^14.0",
     "babel-jest": "^26.0.1",
     "jest": "^26.0.1",
-    "moxios": "^0.4.0",
-    "prettier": "^2.0.5"
+    "prettier": "^2.0.5",
+    "typescript": "^4.5.4"
```
Comment (atdrago) [inline package.json:27]: I added `typescript` as a dev dependency so it's clear what version this project is using.

File: src/http_client.ts [hunk 3]
```diff
@@ -23,13 +17,25 @@ const defaultOptions = {
...
+    name = "NetworkError"
+
+    constructor(
+        message: string,
+        public response: Response,
+        public data: unknown
+    ) {
+        super(message)
+    }
+}
```
Comment (atdrago) [inline src/http_client.ts:30]: This NetworkError class is a little helper that is similar to the type of errors that Axios throws, and allows us to add the response and any data, so they can be used in a catch block. You can see it used on line 102 below.

File: package.json (hunk 1 above)
Comment (chasers) [reply package.json:12]: Yeah that feature was WIP but can you just pull it out?  

People just need to understand that Logflare's backend is schema'd and fields have types, when you send data it needs to match that type.  Some tooling around this (client side schemas) would be better vs trying to typecast all the time.

**Pull request merged**